src/c/gen.rs

```diff
@@ -0,0 +1,274 @@
+// Copyright 2023, Offchain Labs, Inc.
+// For licensing, see https://github.com/OffchainLabs/cargo-stylus/blob/main/licenses/COPYRIGHT.md
+
+use alloy_json_abi::{Function, JsonAbi, StateMutability};
+use eyre::bail;
+use serde_json::Value;
+use std::fs;
+use std::io::BufReader;
+use std::{collections::HashMap, fmt::Write};
+use tiny_keccak::{Hasher, Keccak};
+
+fn c_bytearray_initializer(val: &[u8]) -> String {
+    let inner: Vec<_> = val.iter().map(|byte| format!("0x{byte:02x}")).collect();
+    format!("{{{}}}", inner.join(", "))
+}
+
+pub fn c_gen(in_path: String, out_path: String) -> eyre::Result<()> {
+    let f = fs::File::open(&in_path)?;
+
+    let input: Value = serde_json::from_reader(BufReader::new(f))?;
+
+    let Some(input_contracts) = input["contracts"].as_object() else {
+        bail!("did not find top-level contracts object in {}", in_path)
+    };
+
+    let mut pathbuf = std::path::PathBuf::new();
+    pathbuf.push(out_path);
+
+    for (solidity_file_name, solidity_file_out) in input_contracts {
+        let debug_path = vec![solidity_file_name.as_str()];
+        let Some(contracts) = solidity_file_out.as_object() else {
+            println!("skipping output for {:?} not an object..", &debug_path);
+            continue;
+        };
+        pathbuf.push(solidity_file_name);
+        fs::create_dir_all(&pathbuf)?;
+
+        for (contract_name, contract_val) in contracts {
```

Comment:
Hi @tsahee looks like you could use https://docs.rs/alloy-json-abi/latest/alloy_json_abi/ and avoid all of this json parsing. You also get access to state mutability and everything else needed in this file